### PR TITLE
Gen script for Redmine

### PR DESCRIPTION
**Pull request by balcirakpeter**
-Gen script for users, projects and memberships export
-Fixed: Empty containers replaced by strings as key for hashes

### CODE REVIEW

**Comment by balcirakpeter** [comment]
Please close pull request #80

File: gen/redmine.pl [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	my %resourceAttributes = attributesToHash $project->getAttributes;
+
+	my $projectID =	         $resourceAttributes{$A_RESOURCE_PROJECTID};
+	my $projectName =        $resourceAttributes{$A_RESOURCE_PROJECTNAME};
+	my $description =        $resourceAttributes{$A_RESOURCE_DESCRIPTION};
+	my $isPublic =           "True";
+	my @roles =              $resourceAttributes{$A_RESOURCE_ROLE};
+
+	if(exists $projectStruc->{$projectID} && $projectStruc->{$projectID}->{$A_RESOURCE_PROJECTNAME} != $projectName){
```
Comment (licehammer) [inline gen/redmine.pl]: Strings must be compared using "eq" on "ne" operator instead of "==" or "!=".

File: gen/redmine.pl [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		die "Cannot exists more projects with same projectID and with different name! \n";
+	}
+
+	$projectStruc->{$projectID}->{$A_RESOURCE_PROJECTNAME} =  $projectName;
+	$projectStruc->{$projectID}->{$A_RESOURCE_DESCRIPTION} =  $description;
+	$projectStruc->{$projectID}->{'isPublic'} =               $isPublic;
+
+		foreach my $memberData ($project->getChildElements) {
+			processUsers $projectID, $memberData, @roles;
+		}
```
Comment (licehammer) [inline gen/redmine.pl]: Indentation

**Comment by licehammer** [comment]
Merged locally with different commit message.

**Pull request closed without merging**